pychromecast/__init__.py

```diff
@@ -194,6 +197,13 @@ def __init__(self, host, port=None, device=None, **kwargs):
         if blocking:
             self.socket_client.start()
 
+        # Set to True by default if cast_type == CAST_TYPE_GROUP
+        self.dynamic_host = True if self.cast_type == CAST_TYPE_GROUP \
+            else False
+        self.dynamic_host = kwargs.pop('dynamic_host', self.dynamic_host)
+        if self.dynamic_host:
+            self.start_dynamic_host_tracking()
```

Comment:
Thanks for the feedback, the single thread for all groups is a good idea. Had a quick look at it but ran into some issues with the `__del__` -method. It seems like it never gets called. Probably due to using strong (non-weak?) references for the listener registration, see for example [here](https://github.com/balloob/pychromecast/blob/master/pychromecast/__init__.py#L180).